pkg/services/sqlstore/migrations/ualert/tables.go

```diff
@@ -345,6 +345,10 @@ func AddAlertmanagerConfigMigrations(mg *migrator.Migrator) {
 	mg.AddMigration("add configuration_hash column to alert_configuration", migrator.NewAddColumnMigration(alertConfiguration, &migrator.Column{
 		Name: "configuration_hash", Type: migrator.DB_Varchar, Nullable: false, Default: "'not-yet-calculated'", Length: 32,
 	}))
+
+	mg.AddMigration("add successfully_applied column to alert_configuration", migrator.NewAddColumnMigration(alertConfiguration, &migrator.Column{
+		Name: "successfully_applied", Type: migrator.DB_Bool, Nullable: false, Default: "0",
```

Comment:
I know I suggested this name originally, so sorry for not suggesting this earlier, and if we don't want to do this we can choose not to.

There's a blog post on DB design that I read a while back, that always stuck with me: [You might as well timestamp it](https://changelog.com/posts/you-might-as-well-timestamp-it)

WDYT about instead of storing a bool, storing the timestamp at which the configuration was applied? This column can become a nullable int or nullable datetime instead. Timestamp non-null becomes the same as `true` today.

This can be useful information to show in the UI, to help the user pick the right config. Even if we don't, it's useful in escalations, as it's easier to identify which config is which. The extra space is negligible, as this table only ever stores max 200 rows anyway, so increasing the column from 1 byte to 4 bytes is only 600 extra bytes of storage.

Reply:
We need the timestamp anyway, so it's a great thing to add it here!